src/screens/ENSSearchSheet.js

```diff
@@ -72,18 +73,20 @@ export default function ENSSearchSheet() {
   }, [isAvailable, isInvalid, isRegistered]);
 
   const handlePressContinue = useCallback(() => {
-    startRegistration(`${searchQuery}${ENS_DOMAIN}`);
+    startRegistration(`${searchQuery}${ENS_DOMAIN}`, REGISTRATION_MODES.CREATE);
     Keyboard.dismiss();
     navigate(Routes.ENS_ASSIGN_RECORDS_SHEET);
   }, [navigate, searchQuery, startRegistration]);
 
-  useEffect(() => {
-    debouncedSearchQuery.length > 3 &&
-      startRegistration(
-        `${debouncedSearchQuery}${ENS_DOMAIN}`,
-        REGISTRATION_MODES.CREATE
-      );
-  }, [debouncedSearchQuery, startRegistration]);
+  useFocusEffect(
```

Comment:
ah nice yeah this makes sense, i tried to repro the bug where you added a record in ENSAssignRecordsSheet and then go back to search, the record is cleaned up correctly if the name is changed